src/Microsoft.NetCore.Analyzers/Core/Runtime/SpecifyIFormatProvider.cs

```diff
@@ -129,10 +131,15 @@ public override void Initialize(AnalysisContext analysisContext)
                     var targetMethod = invocationExpression.TargetMethod;
 
                     #region "Exceptions"
+                    const string ToStringMethodName = "ToString";
                     if (targetMethod.IsGenericMethod || targetMethod.ContainingType == null || targetMethod.ContainingType.IsErrorType() ||
                         (targetMethod.ContainingType != null &&
                          (activatorType != null && activatorType.Equals(targetMethod.ContainingType)) ||
-                         (resourceManagerType != null && resourceManagerType.Equals(targetMethod.ContainingType))))
+                         (resourceManagerType != null && resourceManagerType.Equals(targetMethod.ContainingType)) ||
+                         (targetMethod.Name == ToStringMethodName &&
```

Comment:
I needed to add the method name check for the `string` type so I thought that it was better to also use it for `bool` and `char`.